File: src/Systems/Fortran/patterns/iterator.f90
module iterator_example
  implicit none
contains
  logical function run()
    integer :: values(3), seen(3), i
    values = [10, 20, 30]
    seen = 0
    do i = 1, size(values)
      seen(i) = values(i)
    end do
    run = all(seen == values)
  end function
end module

File: src/Systems/Swift/patterns/Iterator.swift
enum IteratorExample {
    static func run() -> Bool {
        var iterator = [10, 20, 30].makeIterator()
        let seen = [iterator.next(), iterator.next(), iterator.next()].compactMap { $0 }
        return seen == [10, 20, 30] && iterator.next() == nil
    }
}
